mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,23 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+      var maxIdx = indices(0)
+      var maxValue = values(0)
+
+      foreachActive { (i, v) =>
```

Comment:
The latest commit should have resolved this. If you give the above values to a new sparse vector the max returned will be -1 not 0. I added unit tests to cover this and they appear to be passing so maybe I am totally misunderstanding the bug you are discussing?